libs/core/src/components/sage-checkbox/sage-checkbox.tsx

```diff
@@ -0,0 +1,117 @@
+import { Component, h, Prop, Host, Event, EventEmitter } from '@stencil/core';
+
+@Component({
+  tag: 'sage-checkbox',
+  styleUrl: 'sage-checkbox.scss',
+  shadow: true,
+})
+export class SageCheckbox {
+  /**
+   * Whether or not the checkbox is checked.
+   */
+  @Prop() checked: boolean;
+
+  /**
+   * Whether or not the checkbox is disabled.
+   */
+  @Prop() disabled: boolean;
+
+  /**
+   * Whether or not the checkbox is invalid.
+   */
+  @Prop() invalid: boolean;
+
+  /**
+   * String used for checkbox `id` attribute and label `for` attribute.
+   */
+  @Prop() checkboxId: string;
+
+  /**
+   * String used for label text next to checkbox.
+   */
+  @Prop() label: string;
+
+  /**
+   * String used for message below checkbox.
+   */
+  @Prop() message: string;
+
+  /**
+   * String used for checkbox `name` attribute.
+   */
+  @Prop() name: string;
+
+  /**
+   * Whether or not checkbox should display as indeterminate. Prop is for visual styling only.
+   * Only JavaScript can set the objects `indeterminate` property. See [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#indeterminate_state_checkboxes).
+   */
+  @Prop() indeterminate: boolean;
+
+  /**
+   * Whether or not the checkbox is required.
+   */
+  @Prop() required: boolean;
+
+  /**
+   * The value of the checkbox that is submitted with a form.
+   */
+  @Prop() value: string;
+
+  /**
+   * Emits a boolean indicating whether the checkbox is currently checked or unchecked.
+   */
+  @Event() sageCheckbox: EventEmitter<boolean>;
+
+  private handleCheckboxChange(event: Event) {
+    if (this.disabled) {
+      return;
+    }
+
+    const target = event.target as HTMLInputElement;
+    const isChecked = target.checked;
+
+    this.sageCheckbox.emit(isChecked);
+  }
+
+  private classNames() {
+    let className = `sage-checkbox`;
+
+    if (this.invalid) {
```

Comment:
It's probably fine to append classes this way as it lends itself more to the React side. I'm mine I did the following to achieve this behavior:
```
class={`
            sage-component
            ${this.prop ? (`sage-component--${this.prop}`) : ''}
            ${this.prop ? 'sage-prop' : ''}
          `}

What are your thoughts on this approach?

Reply:
I think I based this on a similar solution in the button component. I think the `markup`/`return` is easier to read, but no real preference here so not 100% either way.

Reply:
I don't have a strong opinion either way. However, I will throw another possible solution into the mix. 😂 

```
const classNames = ['sage-checkbox'];

if (this.invalid) { classNames.push('is-invalid'); }
if (this.indeterminate) { classNames.push('is-indeterminate'); }
if (this.disabled) { classNames.push('is-disabled'); }

return classNames.join('  ');
```

If you decide to keep what you currently have, I would suggest not defining a variable for a single use. e.g const invalidClassname = 'is-valid'. It was longer to type that versus `className += '  is-invalid';`